### PR TITLE
feat(heureka): optimizes performance - separates total count and page info from main data

### PR DESCRIPTION
**Pull request by hodanoori**
# Summary

Optimized the data fetching for main views in Heureka to improve performance with large datasets (millions of issues). This change splits the fetch requests for all three views (IssueMatches, Services, and Components) by separating the total count and page info queries from the main data queries.

# Changes Made

- Split the GraphQL queries for each view (IssueMatches, Services, Components) to separate fetching of total count and page info from the main data.
- Updated the shared ListController to send two requests per view: one for fetching the main data quickly, and another for fetching total count and page info, which takes longer.
- Set up queryFn for both main and count requests in useQueryClientFn to handle each query independently.
- Added CCRN to the IssueMatches list view.
- Added issue description to the issue details panel.

Some small fixes:
- Automatically close the details panel when switching between different views.

# Related Issues

- closes https://github.com/cloudoperators/heureka/issues/244
- closes https://github.com/cloudoperators/heureka/issues/185
- closes https://github.com/cloudoperators/heureka/issues/245

# Screenshots (if applicable)

# Testing Instructions

1. `npm i`
2. `npm run TASK`

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [ ] My changes generate no new warnings or errors.

### CODE REVIEW

File: apps/heureka/src/components/issueMatches/IssueMatchesDetails.jsx [hunk 1]
```diff
@@ -36,6 +36,13 @@ const IssueMatchesDetails = () => {
               <LoadElement elem={issue?.issue?.primaryName} />
             </DataGridCell>
           </DataGridRow>
+          <DataGridRow>
+            <DataGridHeadCell>Description</DataGridHeadCell>
+            <DataGridCell>
+              {/* // Take description from the first issueVariant, because if there are multiple, they have the same priority (edge case). */}
+              <LoadElement elem={issue?.effectiveIssueVariants?.edges?.[0]?.node?.description} />
```
Comment (barsukov) [inline apps/heureka/src/components/issueMatches/IssueMatchesDetails.jsx:43]: Could yo not extract this one in a variable above that represent a proper name and delete then a comment please?

**Review by barsukov — COMMENTED** [review]
Little comment from my side

**Comment by andypf** [comment]
I try to search by regular expression, but it seems not to work.

**Comment by hodanoori** [comment]
> I try to search by regular expression, but it seems not to work.

The search in the Heureka API is based on wildcard search, which is simpler but more limited than regular expressions. Therefore, I believe it does not fully support regular expression syntax.

File: apps/heureka/src/components/filters/FilterSelect.jsx [hunk 2]
```diff
@@ -50,8 +50,9 @@ const FilterSelect = ({ entityName, isLoading, filterLabels, filterLabelValues,
           disabled={!filterLabelValues[filterLabel]?.length}
           className="filter-value-select w-96 bg-theme-background-lvl-0"
         >
-          {filterLabelValues[filterLabel] //Ensure already selected values are not displayed in filterValue drop down to avoid duplicate selections
-            ?.filter((value) => !activeFilters[filterLabel]?.includes(value)) // Filter out values that are already active
+          {/* Ensure already selected values are not displayed in filterValue drop down */}
```
Comment (ArtieReus) [inline apps/heureka/src/components/filters/FilterSelect.jsx:53]: I would consider here also to extract the logic to variable and remove the comments from the return statement. Something like this:
```
const filteredOptions = filterLabelValues?.[filterLabel] // Safely access filterLabelValues
  ?.filter((value) => !activeFilters?.[filterLabel]?.includes(value)); // Safely access activeFilters and filter out already active values
```

and then use it in the return statement:

```
    {filteredOptions?.map((value) => (
      <SelectOption value={value} key={value} />
    ))}
```

**Review by ArtieReus — CHANGES_REQUESTED** [review]

**Comment by andypf** [comment]
> > I try to search by regular expression, but it seems not to work.
> 
> The search in the Heureka API is based on wildcard search, which is simpler but more limited than regular expressions. Therefore, I believe it does not fully support regular expression syntax.

ok, but the placeholder is "Search term or regular expression" :)

**Review by ArtieReus — APPROVED** [review]
lgtm

**Review by andypf — APPROVED** [review]

**Review by hgw77 — APPROVED** [review]

**Pull request merged**